File: fraud-service/app/domain/entities/inbox_message.py
from app.domain.entities.entity_base import EntityBase
from datetime import datetime, timezone
from uuid import uuid4

class InboxMessage(EntityBase):
    event_id: str
    processed_at: datetime

    @classmethod
    def create(
        cls,
        event_id: str,
    ) -> "InboxMessage":
        return cls(
            event_id = event_id,
            processed_at = datetime.now(timezone.utc)
        )
        

File: order-service/OrderService/order-service.Domain/Entities/InboxMessage.cs
namespace order_service.Domain.Entities;

public class InboxMessage : EntityBase
{
    public string EventId { get; private set; } = string.Empty;
    public DateTime ProcessedAt { get; private set; }

    private InboxMessage() {}

    public InboxMessage(string eventId)
    {
        EventId = eventId;
        ProcessedAt = DateTime.UtcNow;
    }
}
